google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/DefaultCloudSdkService.java

```diff
@@ -40,4 +83,206 @@ public String getCloudSdkHomePath() {
+
+  @NotNull
+  @Override
+  // TODO this path is incorrect. We need to determine an alternate strategy for loading these libs
+  // maybe from maven central
+  public VirtualFile[] getOrmLibSources() {
+    List<VirtualFile> roots = new ArrayList<>();
+    if (getJavaToolsBasePath() != null) {
+      final File libsDir = getJavaToolsBasePath().resolve(Paths.get("src", "orm")).toFile();
+      final File[] files = libsDir.listFiles();
+      if (files != null) {
+        for (File file : files) {
+          final String url = VfsUtil.getUrlForLibraryRoot(file);
+          final VirtualFile zipRoot = VirtualFileManager.getInstance().findFileByUrl(url);
+          if (zipRoot != null && zipRoot.isDirectory()) {
+            String fileName = file.getName();
+            final String srcDirName = StringUtil.trimEnd(fileName, "-src.zip");
+            final VirtualFile sourcesDir = zipRoot.findFileByRelativePath(srcDirName + "/src/java");
+            if (sourcesDir != null) {
+              roots.add(sourcesDir);
+            } else {
+              roots.add(zipRoot);
+            }
+          }
+        }
+      }
+    }
+    return VfsUtilCore.toVirtualFileArray(roots);
+  }
+
+  @Nullable
+  @Override
+  public File getWebSchemeFile() {
+    return getJavaToolsBasePath() != null
+        ? getJavaToolsBasePath().resolve(Paths.get("docs", "appengine-web.xsd")).toFile()
+        : null;
+  }
+
+  @Override
+  public void patchJavaParametersForDevServer(@NotNull ParametersList vmParameters) {
+    if (getJavaToolsBasePath() != null) {
+      File agentPath = getJavaToolsBasePath().resolve(
+          Paths.get("lib", "agent", "appengine-agent.jar")).toFile();
+      if (agentPath.exists()) {
+        vmParameters.add("-javaagent:" + agentPath.getAbsolutePath());
+      }
+      File patchPath = getJavaToolsBasePath().resolve(
+          Paths.get("lib", "override", "appengine-dev-jdk-overrides.jar")).toFile();
+      if (patchPath.exists()) {
+        vmParameters.add("-Xbootclasspath/p:" + patchPath.getAbsolutePath());
+      }
+    }
+  }
+
+  @Nullable
+  @Override
+  public String getVersion() {
+    return getToolsApiJarFile() != null
+        ? JarUtil.getJarAttribute(
+            getToolsApiJarFile(),
+            "com/google/appengine/tools/info/", // The trailing slash is needed so don't use Path
+            Attributes.Name.SPECIFICATION_VERSION)
+        : null;
+  }
+
+  @Override
+  public boolean isValid() {
+    return !StringUtil.isEmpty(getCloudSdkHomePath())
+        && getToolsApiJarFile() != null
+        && getToolsApiJarFile().exists();
+  }
+
+  private static String findLatestVersion(File dir) {
+    String[] names = dir.list();
+    if (names != null && names.length > 0) {
+      String max = Collections.max(Arrays.asList(names));
+      return FileUtil.toSystemIndependentName(new File(dir, max).getAbsolutePath());
+    }
+    return null;
+  }
+
+  private Map<String, Set<String>> loadBlackList() throws IOException {
```

Comment:
No. My original explanation of all this was incorrect:

There are two separate concepts, a _class_ whitelist (which the common lib is providing us and was previously pulled from the old sdk jar), and a _method_ blacklist which blacklists some methods on classes in the whilelist (for example `System` is whitelisted, but `System#exit` is blacklisted).

Its this blacklist that I don't know where JB got it from. It's a plain text file in the resources folder.